components/Search/Search.tsx

```diff
@@ -1,63 +1,113 @@
+import React, { useEffect, useRef, createElement, Fragment } from "react";
 import styles from "./Search.module.css";
+import { autocomplete } from "@algolia/autocomplete-js";
+import "@algolia/autocomplete-theme-classic";
+import { render } from "react-dom";
+import { debounced } from "utils/debounced";
 
 export interface SearchProps {
   id?: string;
   version?: string;
   className?: string;
 }
 
-const ALGOLIA_HITS = 5 * docVersions.versions.length;
-const CURRENT_VERS = docVersions.versions.find((vers) => vers.latest).name;
+export const getSearchResults = async (query: string) => {
+  try {
+    const rawResponse = await fetch(`/docs/api/search/?query=${query}`, {
+      method: "GET",
+    });
+
+    const response = await rawResponse.json();
+    return response.map((res) => ({ ...res, label: res.title }));
+  } catch (e) {
+    console.error(e);
+  }
+};
+
+const SearchAutocomplete = (props) => {
+  const containerRef = useRef(null);
 
-const Search = ({
-  id = "search",
-  version,
-  className,
-  ...props
-}: SearchProps) => {
-  // docsearch.js is using "window" inside, so it will break ssr if we import it directly
   useEffect(() => {
-    import("docsearch.js").then(({ default: docsearch }) => {
-      docsearch({
-        apiKey: process.env.NEXT_PUBLIC_DOCSEARCH_API_KEY,
-        indexName: "goteleport",
-        inputSelector: `[data-docsearch-input="${id}"]`,
-        debug: false,
-        algoliaOptions: {
-          hitsPerPage: ALGOLIA_HITS,
-        },
-        transformData: function (hits) {
-          return hits
-            .filter((hit) => {
-              if (CURRENT_VERS === version) {
-                return !hit.url.includes("/ver/");
-              }
-
-              return hit.url.includes(version);
-            })
-            .slice(0, 5);
-        },
-      });
+    if (!containerRef.current) {
+      return undefined;
+    }
+
+    const search = autocomplete({
+      container: containerRef.current,
+      renderer: { createElement, Fragment },
+      render({ children }, root) {
+        render(children, root);
+      },
+      ...props,
     });
-  }, [id, version]);
+
+    return () => {
+      search.destroy();
+    };
+  }, [props]);
+
+  return <div className={styles["wrapper-autocomplete"]} ref={containerRef} />;
+};
+
+function ProductItem({ hit }) {
+  let foundHeader = "";
+  let foundContent = "";
+
+  if (hit._snippetResult.headers?.length) {
+    foundHeader = hit._snippetResult.headers[0].value;
+  } else if (hit._snippetResult.content) {
+    foundContent = hit._snippetResult.content.value;
+  }
 
   return (
-    <>
-      <div className={cn(styles.wrapper, className)} {...props}>
-        <Icon name="magnify" className={styles.icon} />
-        <input
-          className={styles.input}
-          type="text"
-          placeholder="Search docs..."
-          data-docsearch-input={id}
-        />
+    <a href={hit.objectID} className="aa-ItemLink">
```

Comment:
Q: Why some of these classes are strings and some use `css-modules`? Is is some requirement of Algolia? Maybe move all code for the elements we have an access to to `css-modules`?

Reply:
In the tags where css modules are used, these are my Algolia style changes via css cascading.
In the string there are library related classes only - algolia provides as classes like `aa-ItemLink`.